handler/src/main/java/io/netty/handler/ssl/OpenSsl.java

```diff
@@ -61,29 +61,25 @@
         // If in the classpath, try to load the native library and initialize netty-tcnative.
         if (cause == null) {
             try {
-                String os = normalizeOs(SystemPropertyUtil.get("os.name", ""));
-                String arch = normalizeArch(SystemPropertyUtil.get("os.arch", ""));
-
-                Set<String> libNames = new LinkedHashSet<String>(3);
-                // First, try loading the platform-specific library. Platform-specific
-                // libraries will be available if using a tcnative uber jar.
-                libNames.add("netty-tcnative-" + os + '-' + arch);
-                if (LINUX.equalsIgnoreCase(os)) {
-                    // Fedora SSL lib so naming (libssl.so.10 vs libssl.so.1.0.0)..
-                    libNames.add("netty-tcnative-" + os + '-' + arch + "-fedora");
-                }
-                // finally the default library.
-                libNames.add("netty-tcnative");
-
-                NativeLibraryLoader.loadFirstAvailable(SSL.class.getClassLoader(),
-                        libNames.toArray(new String[libNames.size()]));
-
-                Library.initialize("provided");
-                SSL.initialize(null);
+                // The JNI library was not already loaded. Load it now.
+                loadTcNative();
             } catch (Throwable t) {
                 cause = t;
                 logger.debug(
-                        "Failed to load netty-tcnative; " +
+                    "Failed to load netty-tcnative; " +
+                        OpenSslEngine.class.getSimpleName() + " will be unavailable, unless the " +
+                        "application has already loaded the symbols by some other means. " +
+                        "See http://netty.io/wiki/forked-tomcat-native.html for more information.", t);
+            }
+
+            try {
+                initializeTcNative();
```

Comment:
nit: cause needs to be set back to null if this succeeds right?  Other code in this file depend on the cause being null to mean success.